realm/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -402,70 +402,70 @@ public TableQuery notEqualTo(long columnIndex[], String value){
     }
     protected native void nativeNotEqual(long nativeQueryPtr, long columnIndex[], String value, boolean caseSensitive);
 
-    public TableQuery beginsWith(long columnIndex, String value, boolean caseSensitive){
+    public TableQuery beginsWith(long columnIndices[], String value, boolean caseSensitive) {
```

Comment:
If we have both methods, they should be used in `RealmQuery`. This means that we have doubled the code paths to test (and we have many predicates in `RealmQuery`). The proposed change is simply making the API for `RealmQuery` more consistent as `contains`, `beginsWith`, and `endsWith` support link queries without any exception.